Mathlib/Analysis/SpecialFunctions/Log/NegMulLog.lean

```diff
@@ -52,18 +52,108 @@ lemma hasDerivAt_mul_log {x : ℝ} (hx : x ≠ 0) : HasDerivAt (fun x ↦ x * lo
   refine DifferentiableOn.differentiableAt differentiableOn_mul_log ?_
   simp [hx]
 
-lemma deriv2_mul_log {x : ℝ} (hx : x ≠ 0) : deriv^[2] (fun x ↦ x * log x) x = x⁻¹ := by
+-- TODO Maybe replace by shorter argument or put in different file?
+lemma abs_min_lt_of_nonneg {α : Type} [LinearOrderedAddCommGroup α] {L R x : α}
+    (hL : 0 ≤ L) (hR : 0 ≤ R) (Llx : L < x) :
+    |min L R| < x := by
+  by_cases xlt : L < R
+  · rw [min_eq_left_of_lt xlt, abs_eq_self.mpr hL]
+    exact Llx
+  · simp only [min_eq_right, le_of_not_lt xlt]
+    rw [abs_eq_self.mpr hR]
+    exact lt_of_le_of_lt (le_of_not_lt xlt) Llx
+
+-- helper lemma for `not_eventually_bounded_zero_mul_log`
+-- Is using `protected` good here? Probably nobody will ever want to use this?
```

Comment:
Are you meaning to use `private`

Reply:
Maybe. What would you suggest? Using `private`? Or inlining the lemma? Or do I *have* to find a much shorter proof?

Reply:
I think using `private` is fine here